src/openms/source/TRANSFORMATIONS/FEATUREFINDER/EmgFitter1D.cpp

```diff
@@ -309,6 +309,69 @@ namespace OpenMS
     //width_ = weighted_sd * std::sqrt(std::pow(1. - (weighted_skew / 2.), 2./3.));
   }
 
+  void EmgFitter1D::setInitialParameters_(const RawDataArrayType& set)
+  {
+    if (param_.getValue("init_mom").toBool())
+    {
+      setInitialParametersMOM_(set);
+      return;
+    }
+
+    // sum over all intensities
+    CoordinateType sum = 0.0;
+    for (Size i = 0; i < set.size(); ++i)
+      sum += set[i].getIntensity();
+
+    // calculate the median
+    Size median = 0;
+    float count = 0.0;
+    for (Size i = 0; i < set.size(); ++i)
```

Comment:
I think we have some code for median calculation in Math. Check if it can be reused?

Reply:
Ah it is a weighted median. I checked.